src/win/process.c

```diff
@@ -970,6 +971,31 @@ int uv_spawn(uv_loop_t* loop,
     }
   }
 
+  /* If cwd is too long, shorten it */
+  if (cwd_len >= MAX_PATH) {
+    WCHAR* long_cwd = cwd;
+    DWORD short_cwd_len, r;
+    short_cwd_len = GetShortPathNameW(long_cwd, NULL, 0);
+    if (short_cwd_len == 0) {
+      err = GetLastError();
+      goto done;
+    }
+
+    cwd = (WCHAR*) uv__malloc(short_cwd_len * sizeof(WCHAR));
```

Comment:
can't we allI think you could avoid this by having a stack allocated array of MAX_PATH size.

Reply:
The counterargument is that `wchar_t[MAX_PATH]` is kind of biggish for an on-stack buffer (64 kB) but we do that elsewhere so I suppose it's okay? It's definitely easier to reason about.